app/src/main/java/org/mozilla/tv/firefox/channels/ChannelTile.kt

```diff
@@ -85,9 +87,27 @@ data class ChannelTile(
 sealed class ImageSetStrategy {
     abstract operator fun invoke(imageView: ImageView)
 
+    protected var transformation: Transformation? = null
+
+    // This cannot be done during strategy instantiation because it will often require
+    // information about the target ImageView (particularly its size)
+    fun setTransformation(transformation: Transformation): ImageSetStrategy {
+        this.transformation = transformation
+        return this
+    }
+
+    protected fun RequestCreator.applyTransformationIfNotNull(transformation: Transformation?): RequestCreator = this
```

Comment:
nit: Can this function be simplified for readability to 
```
protected fun RequestCreator.applyTransformationIfNotNull(transformation: Transformation?): RequestCreator {
    transformation?.let { transform(it) }
    return this
}
```

Reply:
There is one small difference between those two code blocks. In the suggested version, we don't return the result of transform. Usually the Java builder pattern uses a mutable builder, so it would do the same thing, but it isn't necessarily guaranteed to.  We can change it to this, though, which I think preserves the intent of your request:

```kotlin
protected fun RequestCreator.applyTransformationIfNotNull(transformation: Transformation?): RequestCreator {
    transformation?.let { return transform(it) }
    return this
}

```